Source/WebKit/Platform/cocoa/MaterialHostingSupport.swift

```diff
@@ -0,0 +1,109 @@
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if !os(visionOS) && canImport(SwiftUI, _version: 7.0.27)
+
+#if canImport(UIKit)
+@_spi(Private) @_spi(ForUIKitOnly) internal import SwiftUI
+#else
+@_spi(Private) @_spi(ForAppKitOnly) internal import SwiftUI
+#endif
+
+#if canImport(UIKit)
+
+struct MaterialHostingContentViewWrapper: UIViewRepresentable {
+    let contentView: UIView
+
+    func makeUIView(context: Context) -> UIView {
+        contentView
+    }
+
+    func updateUIView(_ uiView: UIView, context: Context) { }
+}
+
+#endif
+
+@objc(WKMaterialHostingSupport)
+class MaterialHostingSupport: NSObject {
+    fileprivate static let materialHostingContentLayerKey = "_materialHostingContentLayerKey"
+
+    @objc
+    class func createHostingLayer() -> CALayer {
+        let contentLayer = CALayer()
+
+        let hostingLayer = CAHostingLayer(rootView: AnyView(_CALayerView(type: CALayer.self) { layer in
+            layer.addSublayer(contentLayer)
+        }))
+
+        hostingLayer.setValue(contentLayer, forKeyPath: materialHostingContentLayerKey)
+
+        return hostingLayer
+    }
+
+    @objc(updateHostingLayer:cornerRadius:)
+    class func updateHostingLayer(_ layer: CALayer, cornerRadius: CGFloat) {
+        guard let hostingLayer = layer as? CAHostingLayer<AnyView> else {
+            return;
+        }
+
+        guard let contentLayer = hostingLayer.value(forKeyPath: materialHostingContentLayerKey) as? CALayer else {
+            return;
+        }
+
+        hostingLayer.rootView = AnyView(_CALayerView(type: CALayer.self) { layer in
+            layer.addSublayer(contentLayer)
+        }.materialEffect(Material._hostedBlurMaterial, in: .rect(cornerRadius: cornerRadius)))
+    }
+
+    @objc(contentLayerForMaterialHostingLayer:)
+    class func contentLayerForMaterialHostingLayer(_ layer: CALayer) -> CALayer? {
+        guard let contentLayer = layer.value(forKeyPath: materialHostingContentLayerKey) as? CALayer else {
+            return nil;
+        }
+
+        return contentLayer;
+    }
+
+#if canImport(UIKit)
+
+    @MainActor @objc(createHostingView:)
```

Comment:
MainActor shouldn't be needed

Reply:
It is needed.

```
error: main actor-isolated property 'rootView' can not be mutated from a nonisolated context
```

Reply:
Removed after specifying `NS_SWIFT_UI_ACTOR` in the header.